aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -117,14 +117,16 @@ public void initialize(Map<String, String> properties) {
     }
 
     // Report Hadoop metrics if Hadoop is available
+    String metricsImpl = properties.getOrDefault(
```

Comment:
can we move this chunk of initialization logic to `CatalogUtil` next to `loadFileIO`?